### PR TITLE
Royalty update fix

### PR DESCRIPTION
**Pull request by tasiov**

### CODE REVIEW

**Review by jhernandezb — CHANGES_REQUESTED** [review]
- add constants for harcoded values
- allow decreasing royalties without percent check

File: contracts/collections/sg721-base/src/contract.rs [hunk 1]
```diff
@@ -221,35 +225,45 @@ where
...
+                        )));
+                    }
+                    if new_royalty_info.share > Decimal::percent(MAX_ROYALTY_SHARE_PCT) {
+                        return Err(ContractError::InvalidRoyalties(
+                            format!("Share cannot be greater than {}%", MAX_ROYALTY_SHARE_PCT)
+                                .to_string(),
+                        ));
+                    }
+                } else {
                 }
```
Comment (jhernandezb) [inline contracts/collections/sg721-base/src/contract.rs:260]: let's remove this else

File: contracts/collections/sg721-base/src/contract.rs [hunk 2]
```diff
@@ -366,40 +380,50 @@ where
...
     }
 }
 
 pub fn share_validate(share: Decimal) -> Result<Decimal, ContractError> {
     if share > Decimal::one() {
-        return Err(ContractError::InvalidRoyalties {});
+        return Err(ContractError::InvalidRoyalties(
+            "Share cannot be greater than 100%".to_string(),
+        ));
     }
```
Comment (jhernandezb) [inline contracts/collections/sg721-base/src/contract.rs:427]: Decimal type allows only positive numbers right ?

**Review by jhernandezb — APPROVED** [review]

File: contracts/collections/sg721-base/src/contract.rs (hunk 2 above)
Comment (tasiov) [reply contracts/collections/sg721-base/src/contract.rs:427]: yes Decimal is Uint under the hood

Screenshot 2023-07-31 at 11 07 18 AM

**Pull request merged**